regcomp.c

```diff
@@ -14314,7 +14314,7 @@ S_handle_user_defined_property(pTHX_
                         s = e;
                     }
                     if (SvCUR(msg) > 0) sv_catpvs(msg, "; ");
-                    sv_catpv(msg, overflow_msg);
+                    sv_catpvs(msg, "Code point too large in \"");
```

Comment:
I object to taking a common value and making it into duplicate copies

Reply:
I think overflow_msg could be changed to a define, and then this code could use the define. Alternatively cant these cases use sv_catpvn() and use sizeof on the constant string (or do i misremember that that works).

Reply:
The code as written, prohibits de-duping identical RO string C literals. ```"foo"``` token de-dupes in the linker. ```static const char foo [] = "foo";```does not and is a violation of C if numeric pointer at ```static symbol foo``` in ```A.o```, is equal to numeric pointer at ```static symbol foo``` in ```B.o```.  ISO C committee argues some people use abs pointers of static fns or static char arrays, as GUIDs or "plugin build numbers" in APIs, or session handles. Or separate "my objects" from "foreign objects", based on a callback fn ptr in that malloc object, and a static callback fn, that is identical src code and identical machine code, in 2 different .o files, with 2 copies of machine code and 2 different pointers. And each .o gets a global "destroy" event, chained somehow, then compares callback fn ptr in malloc object to their private static fn ptr symbol, if same, call "my destructor", else pass malloc obj back to the root API for some other dtor.

CC LTO can do some tricks, usually static failed to inline FNs, but moment you do "&" on a static fns, and static const char arrays, %99 chance, & of will be done on a static const char array, and that ptr to static gets passed to another FN, therefore LTO de-duping is not possible.

```#define``` macro, lower case, or upper case token name, or literal ```"foo"``` token, is only way out of this in C.

 in Makes more sense in the world of closed source, commercial .o files than FOSS world.

For example instead